homeassistant/components/tplink/__init__.py

```diff
@@ -37,14 +44,22 @@ def discover():
     """Set up the TP-Link component."""
     conf = config.get(DOMAIN)
 
+    # Some parts of the config are optional, so we need to initialize the
+    # missing ones now for async_setup_entry
     if conf is not None:
-        hass.data[DOMAIN] = TPLINK_SCHEMA(conf)
+        if CONF_LIGHT not in conf:
+            conf[CONF_LIGHT] = []
```

Comment:
We can put the empty lists as default values in the config schema. This will populate defaults when we have a config yaml.